Plugins/GitUIPluginInterfaces/Settings/StringSetting.cs

```diff
@@ -62,7 +62,7 @@ public override void LoadSetting(ISettingsSource settings, TextBox control)
                 }
 
                 control.Text = control.Multiline
-                    ? settingVal?.Replace("\n", Environment.NewLine)
+                    ? settingVal?.Replace(Environment.NewLine, "\n").Replace("\n", Environment.NewLine)
```

Comment:
Why are we doing this at all?
What is the use-case we are trying to address here?

Reply:
I didn't investigate too much but it bite more than 2 times. Enough to push me to make this fix.

If the multi-lines value of a setting is stored with a "\r\n", the actual code introduce a bad behavior. It transforms this every "\r\n" in "\r\r\n" which is displayed everywhere as 2 end of lines. And every times the setting is saved, this bad behavior continues adding new end of lines.

I don't remember the goal of the original `.Replace()` (7fa0eccc9f007cfe8fe277b210fc44af0ad1e6ba). I imagine it's because we can't guarantee the end of line character especially when the setting is saved in the repository but the strategy I made in the past was not perfect.

This fix could be summed up as "convert only the unix end of line and don't mess the windows ones".

Reply:
```suggestion
                    ? settingVal?.Replace(Environment.NewLine, "\n")?.Replace("\n", Environment.NewLine)
```

Reply:
@mstv I was not 100% sure about that so I tested it yesterday. And resharper seems to tell me that it is not needed.

I imagine that if the string is null, it returns the null value immediately. Otherwise, if that's not null, the string could not become null.

Reply:
Perfect.

Reply:
> if the string is null, it returns the null value immediately. Otherwise, if that's not null, the string could not become null.

That's correct

Reply:
> If the multi-lines value of a setting is stored with a "\r\n", the actual code introduce a bad behavior. It transforms this every "\r\n" in "\r\r\n" which is displayed everywhere as 2 end of lines. And every times the setting is saved, this bad behavior continues adding new end of lines.



> This fix could be summed up as "convert only the unix end of line and don't mess the windows ones".

Could you please add this as a comment to the code

Reply:
Looks like you've already done it 👍